os/drivers/lcd/mipi_lcd.c

```diff
@@ -399,6 +406,49 @@ static int lcd_setcontrast(FAR struct lcd_dev_s *dev, unsigned int contrast)
 	return OK;
 }
 
+FAR void lcd_put_logo(FAR struct lcd_dev_s *dev)
+{
+	int logo_arr_index = 0;
+	int lcd_data_index = CONFIG_LCD_XRES * (CONFIG_LCD_YRES - LOGO_YRES) + (CONFIG_LCD_XRES - LOGO_XRES);
+	int lcd_data_col_count = 0;
+	FAR struct mipi_lcd_dev_s *priv = (FAR struct mipi_lcd_dev_s *)dev;
+
+	/* Memory optimization applied using Rotation buffer
+	 * If rotation is enabled, then we have two buffers allocated for rotation.
+	 * During bootup, rotation buffer will not be used (No putarea call from application)
+	 * Therefore, the rotation buffer can be safely used for storing logo data.
+	 * 
+	 * If rotation is disabled, then we need to allocate memory for full screen data
+	 * and it will allocate memory to lcd_logo_fullscreen_data buffer.
+	 * 
+	 * If rotation is enabled, then lcd_logo_fullscreen_data contains pointer of rotation buffer */
+#if defined(CONFIG_LCD_SW_ROTATION)
+	lcd_logo_fullscreen_data = lcd_buffer[0];
+#else
+	lcd_logo_fullscreen_data = (uint8_t *)kmm_malloc(CONFIG_LCD_XRES * CONFIG_LCD_YRES * 2 + 1);
+	if (!lcd_logo_fullscreen_data) {
+		lcddbg("ERROR: LCD logo data memory allocation failed\n");
+		return;
+	}
+#endif
+	/* Filling buffer with black color using memset and
+	 * then filling it with logo data on specific index in while loop */
+	memset(lcd_logo_fullscreen_data, LCD_BLACK_VAL, CONFIG_LCD_XRES * CONFIG_LCD_YRES * 2);
+
+	while (logo_arr_index < (LOGO_YRES * LOGO_XRES * 2)) {
+		lcd_logo_fullscreen_data[lcd_data_index] = lcd_logo_raw_data[logo_arr_index++];
+		lcd_logo_fullscreen_data[lcd_data_index + 1] = lcd_logo_raw_data[logo_arr_index++];
+		lcd_data_index += 2;
+		lcd_data_col_count += 1;
+		if (lcd_data_col_count == LOGO_XRES) {
+			lcd_data_index += ((CONFIG_LCD_XRES - LOGO_XRES) * 2);
+			lcd_data_col_count = 0;
+		}
+	}
```

Comment:
Let's merge as it is. But logo pixels are overwritten. For performance, let's find a way to avoid overwritten.

Reply:
ok, I will find some algorithm to avoid overwritten.